test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Add.Proj.Tests
+{
+    public class GivenDotnetAddProj : TestBase
+    {
+        const string TestGroup = "NonRestoredTestProjects";
+        const string ProjectName = "DotnetAddProjProjects";
+
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsUsage(string helpArg)
+        {
+            var cmd = new AddProjCommand().Execute(helpArg);
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("idontexist.sln")]
+        [InlineData("ihave?inv@lid/char\\acters")]
+        public void WhenNonExistingSolutionIsPassedItPrintsErrorAndUsage(string solutionName)
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(setup.TestRoot)
+                .WithSolution(solutionName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.ExitCode.Should().NotBe(0);
+            cmd.StdErr.Should().Contain("Could not find");
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("Invalid.sln")]
+        [InlineData("")]
+        public void WhenInvalidSolutionIsPassedItPrintsErrorAndUsage(string slnName)
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(setup.TestRoot)
+                .WithSolution(slnName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.ExitCode.Should().NotBe(0);
+            cmd.StdErr.Should().Contain($"Invalid solution ");
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Fact]
+        public void WhenNoProjectIsPassedItPrintsErrorAndUsage()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(Path.Combine(setup.TestRoot, "MoreThanOne"))
```

Comment:
This is a good example. The Unit Test now has two specifiers for the actual project under test. What's more, the second specifier requires a Path.Combine to get it right. The concept count in this test is strictly increased because of the TestSetup.